File: 08-08-2025_Longest_Prefix_Suffix.py
class Solution:
    def getLPSLength(self, s: str) -> int:
        n = len(s)
        if n == 0:
            return 0

        lps = [0] * n
        length = 0  # length of the previous longest prefix suffix
        i = 1

        while i < n:
            if s[i] == s[length]:
                length += 1
                lps[i] = length
                i += 1
            else:
                if length != 0:
                    length = lps[length - 1]
                else:
                    lps[i] = 0
                    i += 1

        return lps[-1]

File: 08-08-2025_Longest_Prefix_Suffix.cpp
class Solution {
  public:
    int getLPSLength(string &s) {
        // code here
        int n=s.size();
       vector<int>lps(n,0);
        int len=0,i=1;
        while(i<n){
            if(s[i]==s[len]){
                len++;
                lps[i]=len;
                i++;
            }
            else{
                if(len!=0)
                    len=lps[len-1];
                else{
                    lps[i]=0;
                    i++;
                }
            }
        }
        return lps[n-1];
    }
};